dragonfly/implementations/daq_run_interface.py

```diff
@@ -502,50 +491,67 @@ def _start_data_taking(self, directory, filename):
         result = self.provider.cmd(self.daq_target, 'block_channel', payload=self.payload_channel)
         logger.info('start data taking')
 
-        try:
-            # switching from seconds to milisecons
-            duration = self._run_time*1000.0
-            max_duration = self._max_duration*1000.0
-            logger.info('run duration in ms: {}'.format(duration))
-
-            NAcquisitions = duration/max_duration
+        # switching from seconds to milisecons
+        duration = self._run_time*1000.0
+        max_duration = self._max_duration*1000.0
+        logger.info('run duration in ms: {}'.format(duration))
 
-            if NAcquisitions<=1:
-                psyllid_filename = filename+'.egg'
+        NAcquisitions = duration/max_duration
+        if NAcquisitions<=1:
+            psyllid_filename = filename+'.egg'
+            if not os.path.exists(directory):
+                os.makedirs(directory)
 
+            logger.info('Going to tell psyllid to start the run')
+            payload = {'channel':self.channel_id, 'filename': os.path.join(directory, psyllid_filename), 'duration':duration}
+            try:
+                result = self.provider.cmd(self.psyllid_interface, 'start_run', payload=payload)
+            except:
```

Comment:
We still need to be explicit about what to catch in any except block. For example, if you want to catch any DriplineException that is raised during the `self.provider.cmd` you could do
```python
try:
    result = <all that stuff>
except exceptions.DriplineException as e:
    <whatever logic you need>
```